cli/client/path.go

```diff
@@ -0,0 +1,42 @@
+package client
+
+import (
+	"net/url"
+	"path"
+)
+
+const coreAPIGroup = "core"
+const coreAPIVersion = "v2"
+
+func createBasePath(group, version string, resType ...string) func(...string) string {
+	fn := createNSBasePath(group, version, resType...)
+
+	return func(components ...string) string {
+		// first arg "" represents that no namespace has been specified
+		return fn("", components...)
+	}
+}
+
+func createNSBasePath(group, version string, resType ...string) func(string, ...string) string {
```

Comment:
It might be useful to describe how the returned function works, especially regarding how its first argument works: `""` being a special value that bypass namespaces, ...